main.go

```diff
@@ -537,19 +545,33 @@ func runCommands(commandString string, theProcessType ProcessType, killServer bo
 			// do nothing we continue after kill errors
 			Warning.log("Watcher killProcess received error ", err)
 		}
-		cmd, err = startProcess(commandString, fileWatcher)
-		if err != nil {
 
-			Warning.log("ERROR start process for file watcher received error ", err)
+		commandToUse := commandString
+		processTypeToUse := fileWatcher
+
+		if !killServer {
+			// this path is only relevant for APPSODY_<RUN/DEBUG/TEST>KILL_SERVER=FALSE
+			// get the process of the current server (should not be nil ever) and send benign SIG 0 to the server proces
+			if cmps.processes[server] != nil && cmps.processes[server].Signal(syscall.Signal(0)) != nil {
+				// if there is no server process, an error is returned
+				Debug.log("The server process with pid:", cmps.processes[server].Pid, "was not found, and APPSODY_<action>_KILL is set to false. The server will be restarted.")
+				//start the server with the startCommand, not the watch action command
+				commandToUse = startCommand
```

Comment:
In this code path, I think the server should be started and then the on_change command should be run.
Is the on_change command still going to be run?

Reply:
I am not sure I agree.  So isn't this analogous to the cold start scenario.  

This happens in two current stacks, in the spring case at server start (and only server start).
In the microprofile case, is this also only at server start? 

I think this could introduce several timing issues in the controller code atm.

Would we do APPSODY_RUN and APPSODY_ON_CHANGE at server start up?
If something doesn't compile due to APPSODY_RUN, how would the basic server start scenario work?  I think this would suggest we'd need to redefine exactly what the bounds of APPSODY_RUN are.

If we had to do this, I think what we'd do is create another function, like restartServer(command). It would be spun in its own separate thread with its own start.wait sequence.  I would worry about the timing issues doing the run and the ON_CHANGE might cause since we would kick off two threads at the same time or nearly the same time if we introduce a wait.  How would you really tell if the server is up and you could fire off the ON_CHANGE?  signals, ps -ef wouldn't tell you its up.  you could browse a log file, but that would be specific to a given runtime.

Reply:
Okay I agree the timing would be problematic. I think what you have today is okay (if server process doesn't exist, exec the run server, otherwise exec on_change). Especially since we will encourage stack creators to compile the code in the run server command.